File: 1.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        
        Dictionary<int, int> hashTable = new Dictionary<int, int>(); 
        
        for(int i=0; i<nums.Length; i++){
            
            int difference = target - nums[i];
            
            if(hashTable.ContainsKey(difference)){
                return new int[]{hashTable[difference], i};
            }
            if(!hashTable.ContainsKey(difference)){
                hashTable.TryAdd(nums[i], i);
            }
        }
        
        return new int[]{0, 0};
    }
}

File: 1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const hash = new Map();

    for(let i =0; i<nums.length; i++){
        const n = nums[i];
        const diff = target - n;

        if(hash.has(diff)){
            return [hash.get(diff), i];
        }

        hash.set(n, i);
    }

    return [-1, -1];

};
